fs/base.py

```diff
@@ -310,6 +312,22 @@ def setinfo(self, path, info):
 
         """
 
+    @staticmethod
+    def quote(path_snippet):
+        if _WINDOWS_PLATFORM and ":" in path_snippet:
+            drive_letter, path = path_snippet.split(":", 1)
+            path = path.replace("\\", "/")
+            if six.PY2:
+                path = path.encode("utf-8")
+            path = six.moves.urllib.parse.quote(path)
+            path_snippet = "{}:{}".format(drive_letter, path)
+        else:
+            path_snippet = path_snippet.replace("\\", "/")
```

Comment:
Is this logic correct for non-Windows platforms?
E.g.
```bash
$ touch /tmp/file\\foo:bar
$ ls -l /tmp
-rw-rw-r-- 1 qpc  qpc     0 Aug  8 21:42 file\foo:bar
```

Reply:
No. fixed it.